forte/processors/nltk_processors.py

```diff
@@ -97,6 +102,48 @@ def penn2morphy(penntag: str) -> str:
         return 'n'
 
 
+class NLTKChunker(PackProcessor):
+    r"""A wrapper of NLTK chunker.
+    """
+    def __init__(self):
+        super().__init__()
+        self.chunker = None
+        self.token_component = None
+
+    # pylint: disable=unused-argument
+    def initialize(self, resource: Resources, configs: HParams):
+        self.chunker = RegexpParser(configs.pattern)
+
+    @staticmethod
+    def default_hparams():
+        r"""This defines a basic Hparams structure for NLTKChunker.
+        """
+        return {
+            'pattern': 'NP: {<DT>?<JJ>*<NN>}',
+        }
+
+    def _process(self, input_pack: DataPack):
+        for sentence in input_pack.get(Sentence):
+            token_entries = list(input_pack.get(entry_type=Token,
+                                                range_annotation=sentence,
+                                                component=self.token_component))
+            tokens = [(token.text, token.pos) for token in token_entries]
+            cs = self.chunker.parse(tokens)
+
+            index = 0
+            for chunk in cs:
+                if hasattr(chunk, 'label'):
+                    begin_pos = token_entries[index].span.begin
+                    end_pos = token_entries[index + len(chunk) - 1].span.end
+                    phrase = Phrase(input_pack, begin_pos, end_pos)
+                    kwargs_i = {"phrase_type": chunk.label()}
+                    phrase.set_fields(**kwargs_i)
+                    input_pack.add_or_get_entry(phrase)
+                    index += len(chunk)
+                else:
+                    index += 1
```

Comment:
Is this assuming the len of the chunk is 1? Is it better if we do `index += len(chunk)` here as well?

Reply:
I think this is correct. When the chunk has no `label` attribute, it contains only one token.